container/go/pkg/compat/overrideConfig.go

```diff
@@ -0,0 +1,545 @@
+	empty = hex.EncodeToString(b[:])
+	return
+}
+
+// extractValue returns the contents of a file pointed to by value if it starts with a '@'.
+func extractValue(value string) (string, error) {
+	if strings.HasPrefix(value, "@") {
+		f, err := os.Open(value[1:])
+		defer f.Close()
+		shaHash, err := ioutil.ReadAll(f)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to read content from file %s", value)
+		}
+		return string(shaHash), nil
+	}
+	return value, errors.Wrapf(nil, "unexpected value, got: %s want: @{...}", value)
+}
+
+// keyValueToMap converts an array of strings separated by '=' into a map of key-value pairs.
+func keyValueToMap(value []string) (map[string]string, error) {
+	convMap := make(map[string]string)
+	var temp []string
+	for _, kvpair := range value {
+		temp = strings.Split(kvpair, "=")
+		if len(temp) != 2 {
+			return convMap, errors.New("value in array are not of format key=value")
+		}
+		key, val := temp[0], temp[1]
+		convMap[key] = val
+	}
+	return convMap, nil
+}
+
+// Stamp provides the substitutions of variables inside {} using info in file pointed to
+// by stampInfoFile.
+func Stamp(inp string, stampInfoFile []string) (string, error) {
+	if len(stampInfoFile) == 0 || inp == "" {
+		return inp, nil
+	}
+	formatArgs := make(map[string]interface{})
+	for _, infofile := range stampInfoFile {
+		f, err := os.Open(infofile)
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to open file %s", infofile)
+		}
+		defer f.Close()
+		// scanner reads line by line and discards '\n' character by default.
+		scanner := bufio.NewScanner(f)
+		var temp []string
+		for scanner.Scan() {
+			temp = strings.Split(scanner.Text(), " ")
+			key, val := temp[0], temp[1]
+			if _, ok := formatArgs[key]; ok {
+				fmt.Printf("WARNING: Duplicate value for key %s: using %s", key, val)
+			}
+			formatArgs[key] = val
+		}
+		if err = scanner.Err(); err != nil {
+			return "", errors.Wrapf(err, "failed to read line from file %s", infofile)
+		}
+	}
+
+	o := inp
+	for k, v := range formatArgs {
+		if vStr, ok := v.(string); ok {
+			o = strings.ReplaceAll(o, fmt.Sprintf("{%s}", k), vStr)
+		} else {
+			return "", errors.New("argument to format is not of string type")
+		}
+	}
+	return o, nil
+}
+
+// mapToKeyValue reverses a map to a '='-separated array of strings in {key}={value} format.
+func mapToKeyValue(kvMap map[string]string) []string {
+	keyVals := []string{}
+	for k, v := range kvMap {
+		keyVals = append(keyVals, fmt.Sprintf("%s=%s", k, v))
+	}
+
+	sort.Strings(keyVals)
+	return keyVals
+}
+
+// resolveVariables resolves the environment variables embedded in the given value using
+// provided map of environment variable expansions.
+// It handles formats like "PATH=$PATH:..."
+func resolveVariables(value string, environment map[string]string) (string, error) {
+	var i interface{}
+	i = "this is a string type"
+	var errorMet = false
+	mapper := func(p string) string {
+		switch i.(type) {
```

Comment:
yes that is the purpose. It is a hack for the mapper function that is used in os.Expand because I do not want to loop through all the contents of the environment variables and add each one as a switch statement. So I am using the fact that they're all strings to do so.

Reply:
The above will not work, as it requires the switch statements to work.

Reply:
Not sure what you mean it won't work. This works for me https://play.golang.org/p/f2dZBZini50